### ISSUE TITLE
[BUG] signal 11 (SIGSEGV) native crash

### ISSUE THREAD

**Issue by prok155** [mono/SkiaSharp#1733]
**Description**

Hello. Recently I've decided to update my app and I switched app to androidx. I've published my app to google play and see a lot of native crashes (stacktrace below), mainly on Android 11.  The problem is with libSkiaSharp.so, so I guess that there is something wrong with skiasharp. I can't reproduce that error on emulators.

**Stacktrace**

`signal 11 (SIGSEGV), code 1 (SEGV_MAPERR)`

`*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***`
`pid: 0, tid: 0 >>> com.companyname.pcgname<<<`

`backtrace:`
` #00  pc 000000000007c8a8  /data/app/com.companyname.pcgname-1/lib/arm/libSkiaSharp.so`

**Basic Information**

- Version with issue:  Skiasharp.Views.Forms 2.80.2 
- Last known good version: Skiasharp.Views.Forms 2.80.2 
- IDE:   Visual Studio 2019
- Platform Target Frameworks: 
  - Android: SDK 30

**Comment by DoubleDBE** [comment]
Same here with the same DLL versions mentiod by OP.

This log is taken from an Android 10 emulator, but i have these issues on physical devices as well.

```
07-04 04:33:42.431: A/DEBUG(26429): *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
07-04 04:33:42.431: A/DEBUG(26429): Build fingerprint: 'google/sdk_gphone_x86/generic_x86:10/QSR1.200715.002/6695061:userdebug/dev-keys'
07-04 04:33:42.431: A/DEBUG(26429): Revision: '0'
07-04 04:33:42.431: A/DEBUG(26429): ABI: 'x86'
07-04 04:33:42.432: A/DEBUG(26429): Timestamp: 2021-07-04 04:33:42+0000
07-04 04:33:42.432: A/DEBUG(26429): pid: 26111, tid: 26135, name: Finalizer  >>> com.company.app <<<
07-04 04:33:42.432: A/DEBUG(26429): uid: 10133
07-04 04:33:42.433: A/DEBUG(26429): signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x302015a
07-04 04:33:42.433: A/DEBUG(26429):     eax 0302015a  ebx b7df8004  ecx b7661550  edx c4214e50
07-04 04:33:42.433: A/DEBUG(26429):     edi b219ce08  esi b411bf20
07-04 04:33:42.433: A/DEBUG(26429):     ebp c4214ec8  esp c4214e40  eip b766fcea
07-04 04:33:42.528: A/DEBUG(26429): backtrace:
07-04 04:33:42.528: A/DEBUG(26429):       #00 pc 0006ecea  /data/app/com.company.app-zqpOt9HQ87RSg-NvydKF7A==/lib/x86/libSkiaSharp.so
```

**Comment by prok155** [comment]
Have you managed to reproduce it on emulator? Which line of code causes the crash? Most of my app's crashes are form Android 11, but on android 10 and prevoius it occurs as well.

**Comment by thisisthekap** [comment]
We are experiencing the same issue.

Distribution by Android version and device:
![image](https://user-images.githubusercontent.com/9071974/128730026-59bad3c4-35eb-4063-aa71-c084a78497ae.png)

**Comment by prok155** [comment]
Have you managed to reproduce it? My app still experince this error, I tried to find this error on emulators and I've done a lot of tests on firebase TestLab, but it didnt help me.

**Comment by thisisthekap** [comment]
Unfortunately, the same applies to us.

@mattleibow Are you able to give some light to this issue?

**Comment by thisisthekap** [comment]
Just did some additional research. This issue seems to cause the majority of our Android production crashes. But still unable to reproduce it.

@DoubleDBE Were you able to reproduce this issue?

**Comment by Huaba93** [comment]
We are experiencing the same problem. Android 10 is the most affected. This issue causes 70% of our android crashes and occurs in about 1.8% of the sessions. So it is very critical!

@prok155 are you using `SKCanvasView` or `SKGLView`? If SKGLView, are you using `HasRenderLoop`?

**Comment by prok155** [comment]
Hello. I updated my app in google play store. As I suspected, native crash was caused by incorrect disposing of SKPaint, following code caused an error:

`SKPaint textPaint = new SKPaint();`
`textPaint.Dispose();`

I removed this line of code `textPaint.Dispose()` and error disappeared from google play console. I close this issue, if your app still suffer that error it is probably caused by incorrect disposing and memory management.
